Automattic/pocket-casts-ios
# End of Year: sharing

| 📘 Project: #376 |
|:---:|

Adds text content for each story shared and for the top episode and longest episode add a link to them.

<img src="https://user-images.githubusercontent.com/7040243/199720704-b77606fe-6f08-4fb9-8300-cef66b18492b.png" width="300">

## Sharing content to other apps

How the shared content is handled is up to the app. Some apps work fine with images + text, and some do not. They all behave differently. Here's a list of apps I tested with and how they behave:

1. Tumblr: image + link. If the content doesn't have links, it shares the text.
2. WordPress: image + text
2. DayOne: image + text.
3. Facebook: only image. Facebook doesn't accept text and sharing just the image leaves the user to choose between sharing as a story or post.
4. Twitter: image + text.
5. WhatsApp: only image. If we share any text, the image will be ignored.
6. Telegram: image + text.
7. Messages: image + text without hashtags.

## To test

> **Note**
>
> 📱 For better testing, run this on your real device.

1. Enable `endOfYear` and `tracksLoggingEnabled` flags in `FeatureFlag.swift`
2. Run the app and make sure you're logged in to an account with a few items on Listening History
3. Check your 2022 EoY stats (either by tapping "View My 2022" on the modal or going to Profile and tapping the card there)
4. When the first story appears, tap "Share"
5. ✅ Ensure `end_of_year_story_share ["story": "intro"]` is tracked
6. Share to Notes or Twitter
7. ✅ Check the content and the image for the shared story
8. ✅ Ensure `end_of_year_story_shared ["activity": "com.apple.UIKit.activity.CopyToPasteboard", "story": "intro"]` is tracked, where activity depends on what action you've done
8. Repeat these steps for each story, always checking the events and the output (image + text)

### Additional checks

1. ✅ For the top 1 podcast, check that the URL contained in the text redirects to the podcast
2. ✅ For the longest episode, check that the URL contained in the text redirects to the episode
3. ✅ For the top 5 podcasts, check that the URL opens a list containing exactly the 5 top podcasts

Feel free to test sharing the content to other apps.

## Checklist
